src/qamqpclient.cpp

```diff
@@ -531,7 +536,11 @@ void QAmqpClientPrivate::startOk()
     clientProperties["version"] = QString(QAMQP_VERSION);
     clientProperties["platform"] = QString("Qt %1").arg(qVersion());
     clientProperties["product"] = QString("QAMQP");
+#if QT_VERSION >= 0x060000
+    clientProperties.insert(customProperties);
+#else
     clientProperties.unite(customProperties);
```

Comment:
@mbroadst is it sure that the inital intention of the "unite" was to create duplicate entries for the same key?
I don't know if this is possible to communicate in RabbitMQ because the context is not provided here which key/value pair comes first or from which original source.

Reply:
No, I'm sure that was a bug waiting to happen. Is QHash::insert supported in qt < 6?

Reply:
Yes, it is there since Qt4 or earlier. I will change it and use it always.